### PR TITLE
Wip cppcheck271 useStlAlgorithm part2

### PR DESCRIPTION
**Pull request by isildur-g**
this is part 2 of the useStlAlgorithm cppcheck warnings. 
Most of these are simply comments to ignore the warning, as in almost all
cases replacing a simple operation with a function call will kill performance and make
clear to understand code more muddled. 
we did use it in most of the instances of copy, fill, and accumulate. 
Closes some: #252

### CODE REVIEW

File: examples/patbench.cc [hunk 1]
```diff
@@ -461,6 +463,7 @@ class Sigdata {
         vector<const char *> cstrPatterns;
         cstrPatterns.reserve(patterns.size());
         for (const auto &pattern : patterns) {
+            // cppcheck-suppress useStlAlgorithm
```
Comment (markos) [inline examples/patbench.cc]: this might be a std::copy with std::back_inserter, no?

File: src/fdr/fdr_compile.cpp [hunk 2]
```diff
@@ -136,6 +136,7 @@ void FDRCompiler::createInitialState(FDR *fdr) {
         const vector<LiteralIndex> &bucket_lits = bucketToLits[b];
         u32 min_len = ~0U;
         for (const LiteralIndex &lit_idx : bucket_lits) {
+            // cppcheck-suppress useStlAlgorithm
```
Comment (markos) [inline src/fdr/fdr_compile.cpp]: this could be a std::min_element?

File: src/fdr/fdr_confirm_compile.cpp [hunk 3]
```diff
@@ -300,6 +300,7 @@ setupFullConfs(const vector<hwlmLiteral> &lits,
         if (contains(bucketToLits, b)) {
             vector<hwlmLiteral> vl;
             for (const LiteralIndex &lit_idx : bucketToLits.at(b)) {
+                // cppcheck-suppress useStlAlgorithm
```
Comment (markos) [inline src/fdr/fdr_confirm_compile.cpp:303]: similar, std::copy with std::back_inserter?

File: src/nfa/castlecompile.cpp [hunk 4]
```diff
@@ -675,6 +675,7 @@ set<ReportID> all_reports(const CastleProto &proto) {
 depth findMinWidth(const CastleProto &proto) {
     depth min_width(depth::infinity());
     for (const PureRepeat &pr : proto.repeats | map_values) {
+        // cppcheck-suppress useStlAlgorithm
```
Comment (markos) [inline src/nfa/castlecompile.cpp:678]: std::min_element?

File: src/nfa/castlecompile.cpp [hunk 5]
```diff
@@ -683,6 +684,7 @@ depth findMinWidth(const CastleProto &proto) {
 depth findMaxWidth(const CastleProto &proto) {
     depth max_width(0);
     for (const PureRepeat &pr : proto.repeats | map_values) {
+        // cppcheck-suppress useStlAlgorithm
```
Comment (markos) [inline src/nfa/castlecompile.cpp:687]: std::max_element?

File: examples/patbench.cc (hunk 1 above)
Comment (isildur-g) [reply examples/patbench.cc]: with a transform, but yes, done.

File: src/fdr/fdr_confirm_compile.cpp (hunk 3 above)
Comment (markos) [reply src/fdr/fdr_confirm_compile.cpp:303]: justified elsewhere

File: src/nfa/castlecompile.cpp (hunk 4 above)
Comment (markos) [reply src/nfa/castlecompile.cpp:678]: same

**Review by markos — APPROVED** [review]
I'm still unsure about useStlAlgorithm's usefulness tbh, but approving and we can revisit during refactoring.

**Pull request merged**